### PR TITLE
mgr/orchestrator: disallow_untyped_defs = True 

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]

File: src/pybind/mgr/orchestrator/_interface.py [hunk 1]
```diff
@@ -105,11 +106,14 @@ def wrapper(*args, **kwargs):
     wrapper_copy._cli_command.store_func_metadata(func)  # type: ignore
     wrapper_copy._cli_command.func = wrapper_copy  # type: ignore
 
-    return wrapper_copy
+    return cast(FuncT, wrapper_copy)
 
 
-def _cli_command(perm):
-    def inner_cli_command(prefix):
+class InnerCliCommandCallable(Protocol):
```
Comment (tchaikov) [inline src/pybind/mgr/orchestrator/_interface.py:112]: `Protocol` might not be defined.

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/orchestrator/_interface.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator/_interface.py:112]: added in python 3.8. sigh.

**Comment by sebastian-philipp** [comment]
jenkins test api

**Review by tchaikov — APPROVED** [review]

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -685,7 +685,7 @@ def available(self) -> Tuple[bool, str]:
             return False, 'SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key` or `ceph cephadm generate-key`'
         return True, ''
 
-    def process(self, completions: List[CephadmCompletion]) -> None:
+    def process(self, completions: List[CephadmCompletion]) -> None:  # type: ignore
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py:688]:
```suggestion
    def process(self, completions: List[orchestrator.Completion]) -> None:  # type: ignore
```

interesting, Liskov substitution principle ... use the super class here and implement the `evaluate` attr?

**Review by mgfritch — APPROVED** [review]
nit, but otherwise lgtm!

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-01-20_09:35:12-rados:cephadm-wip-swagner-testing-2021-01-19-1509-distro-basic-smithi/

lots of pull failures today.

**Pull request merged**